BeerProject/src/main/java/wawer/kamil/beerproject/service/impl/BeerServiceImpl.java

```diff
@@ -0,0 +1,191 @@
+    }
+
+    @Override
+    public List<Beer> findAllBeersList() {
+        return beerRepository.findAll();
+    }
+
+    @Override
+    public Beer findBeerByBeerId(Long beerId) throws NoContentException {
+        if (beerRepository.existsBeerByBeerId(beerId)) {
+            return beerRepository.findBeerByBeerId(beerId);
+        } else {
+            log.debug(THE_BEER_BASE_ON_ID_HAS_NOT_BEEN_FOUND, beerId);
+            throw new NoContentException();
+        }
+    }
+
+    //get beers by breweryID
+
+    @Override
+    public Page<Beer> findAllBeersByBreweryIdPage(Long breweryId, Pageable pageable) throws NoContentException {
+        if (breweryRepository.existsBreweryByBreweryId(breweryId)) {
+            Brewery brewery = breweryRepository.findByBreweryId(breweryId);
+            return beerRepository.findAllByBrewery(brewery, pageable);
+        } else {
+            log.debug(THE_BREWERY_BASE_ON_ID_HAS_NOT_BEEN_FOUND, breweryId);
+            throw new NoContentException();
+        }
+    }
+
+    @Override
+    public List<Beer> findAllBeersByBreweryIdList(Long breweryId) throws NoContentException {
+        if (breweryRepository.existsBreweryByBreweryId(breweryId)) {
+            Brewery brewery = breweryRepository.findByBreweryId(breweryId);
+            return beerRepository.findAllByBrewery(brewery);
+        } else {
+            log.debug(THE_BREWERY_BASE_ON_ID_HAS_NOT_BEEN_FOUND, breweryId);
+            throw new NoContentException();
+        }
+    }
+
+    @Override
+    public Beer findProperBeerByBreweryIdAndBeerId(Long breweryId, Long beerId) throws NoContentException {
+        if (breweryRepository.existsBreweryByBreweryId(breweryId)) {
+            if (beerRepository.existsBeerByBeerId(beerId)) {
+                Brewery brewery = breweryRepository.findByBreweryId(breweryId);
+                return beerRepository.findBeerByBreweryAndBeerId(brewery, beerId);
+            } else {
+                log.debug(THE_BEER_BASE_ON_ID_HAS_NOT_BEEN_FOUND, beerId);
+                throw new NoContentException();
+            }
+        } else {
+            log.debug(THE_BREWERY_BASE_ON_ID_HAS_NOT_BEEN_FOUND, breweryId);
+            throw new NoContentException();
+        }
+    }
+
+    //post beers
+
+    @Override
+    public Beer addNewBeerToRepository(Beer beer) {
+        return beerRepository.save(beer);
+    }
+
+    @Override
+    public Beer addNewBeerAssignedToBreweryByBreweryId(Long breweryID, Beer beer) throws NoContentException {
+        return breweryRepository.findById(breweryID)
+                .map(brewery -> {
+                    beer.setBrewery(brewery);
+                    return beerRepository.save(beer);
+                }).orElseThrow(NoContentException::new);
+    }
+
+    //put beers
+
+    @Override
+    public Beer updateBeerByBeerId(Long beerId, Beer beer) throws NoContentException {
+        if (beerRepository.existsBeerByBeerId(beerId)) {
```

Comment:
Optional - nie rób sprawdzenia przez `if` - tyczy się wszedzie gdzie zwracasz pojedynczy obiekt z bazy danych - lepiej robić to Optionalem - zastosuj wszedzie